mm2src/coins/utxo/utxo_block_header_storage/sql_block_header_storage.rs

```diff
@@ -85,6 +87,23 @@ fn get_block_height_by_hash(for_coin: &str) -> Result<String, BlockHeaderStorage
     Ok(sql)
 }
 
+fn get_total_block_headers_from_storage(for_coin: &str) -> Result<String, BlockHeaderStorageError> {
+    let table_name = get_table_name_and_validate(for_coin)?;
+    let sql = format!("SELECT COUNT(*) as blocks_count FROM {};", table_name);
+
+    Ok(sql)
+}
+
+fn remove_block_headers_from_storage(for_coin: &str, limit: i64) -> Result<String, BlockHeaderStorageError> {
+    let table_name = get_table_name_and_validate(for_coin)?;
+    let sql = format!(
+        "DELETE FROM {table_name} WHERE block_height IN
+        (SELECT block_height FROM {table_name} limit {limit});",
```

Comment:
How about
```sql
DELETE FROM {table_name} WHERE block_height < {keep_from_block};
```
where `keep_from_block` is passed instead of limit

Reply:
I think limit is better because it'll will be easier to calculate in most cases e.g when current `headers` in `db` is from `4 - 14` and we need to delete `4` headers to accomodate `4` new `headers`.

Reply:
It's also easy to calculate `keep_from_block` easily, It's `to_block_height - max_stored_block_headers + 1`. So let's say the last retrieved header has height `100` and we want to keep a `max_stored_block_headers = 10`. We will delete any blocks with height less than `91` using this sql statement https://github.com/KomodoPlatform/atomicDEX-API/pull/1585#discussion_r1071261171 and will have in storage 10 block headers. I believe this sql statement ```DELETE FROM {table_name} WHERE block_height < {keep_from_block};``` is much simpler and doesn't require selecting then deleting.